File: 1-two-sum/1-two-sum.cpp
class Solution 
{
public:
    vector<int> twoSum(vector<int>& nums, int target) 
    {
        vector<int> res;
        unordered_map<int, int> m;
        for(int i = 0; i < nums.size(); i++)
        {
            int complement = target - nums[i];
            if(m.find(complement) != m.end())
            {
                res.push_back(i);
                res.push_back(m[complement]);
                return res;
            }
            
            m[nums[i]] = i;
        }
        
        return res;
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums_seen = {}
        for i, n in enumerate(nums):
            if target - n in nums_seen:
                return [nums_seen[target - n], i]
            nums_seen[n] = i
        return -1